draft-ietf-quic-tls.md

```diff
@@ -1575,6 +1018,47 @@ version of QUIC defined in {{QUIC-TRANSPORT}} is used.
 The quic_transport_parameters extension is carried in the ClientHello and the
 EncryptedExtensions messages during the handshake.
 
+While the transport parameters are technically available prior to the
+completion of the handshake, they cannot be fully trusted until the handshake
+completes, and reliance on them should be minimized.
+However, any tampering with the parameters will be detected
+when the handshake completes.
+
+
+## QUIC Max Crypto Data Extension {#max_crypto_data}
+
+When QUIC provides TLS messages via the CRYPTO_HS frame, one TLS message may be
+fragmented across different packets.  Thus TLS might have to buffer data.  TLS
+implementations may choose to limit the data they buffer before the handshake is
+completed and close the connection on receiving too much data.
+
+To communicate the maximum amount of data that TLS will allow to be sent in
+CRYPTO_HS frames, TLS may use the `max_crypto_data` extension, defined as
+follows:
+
+~~~
+   enum {
+      max_crypto_data(27), (65535)
+   } ExtensionType;
+
+   struct {
+      uint32 max_crypto_data;
+   } MaxCryptoData;
+~~~
+
+max_crypto_data:
+
+: The maximum number of bytes that can be sent in CRYPTO_HS frames
+
+The `extension_data` field of the extension contains the MaxCryptoData
+structure.
+
+Implementations SHOULD send this extension.  Receivers do not need to process
+this extension.  If the sender cannot fit the handshake data into this limit,
```

Comment:
I think that it is because this extension in considered an option (the sender-side requirement is SHOULD).

A TLS stack can always hard-fail if it sees too much handshake data. This is just a nice way of communicating that.